theseus/embodied/collision/tests/test_eff_obj_contact.py

```diff
@@ -149,3 +150,37 @@ def test_eff_obj_interesect_errors():
         actual = cost_fn.error()
         expected = outputs["error"][sdf_idx, :]
         assert torch.allclose(actual, expected)
+
+
+def test_eff_obj_variable_type():
+    rng = torch.Generator()
+    rng.manual_seed(0)
+    for batch_size in [1, 10, 100]:
+        obj = create_random_se2(batch_size, rng)
+        eff = create_random_se2(batch_size, rng)
+        origin = th.Variable(torch.randn(batch_size, 2).double())
+        sdf_data = th.Variable(torch.randn(batch_size, 10, 10).double())
+        cell_size = th.Variable(torch.rand(batch_size, 1).double())
+        eff_radius = th.Variable(torch.rand(batch_size, 1).double())
+        cost_weight = th.ScaleCostWeight(1.0)
+        cost_function = th.eb.EffectorObjectContactPlanar(
+            obj, eff, cost_weight, origin, sdf_data, cell_size, eff_radius
+        )
+
+        assert isinstance(cost_function.eff_radius, Variable)
+
+        eff_radius_t = torch.rand(batch_size, 1).double()
+
+        cost_function = th.eb.EffectorObjectContactPlanar(
+            obj, eff, cost_weight, origin, sdf_data, cell_size, eff_radius_t
+        )
+
+        assert isinstance(cost_function.eff_radius, Variable)
+
```

Comment:
Here we can add another check that `cost_function.eff_radius.data is eff_radius_t`.

Reply:
@luisenp While `assert cost_function.eff_radius.data is eff_radius_t` gives error as one is `tensor` and other is `th.Variable` .Is it better to use  `assert torch.allclose(cost_function.eff_radius.data,eff_radius_t)` ?

Reply:
@luisenp when i try to do `assert cost_function.eff_radius.data is eff_radius_t` i get this error which expained above 

```
>           assert cost_function.eff_radius.data is eff_radius_t
E           assert tensor([[0.4160]], dtype=torch.float64) is tensor([[0.4160]], dtype=torch.float64)
E            +  where tensor([[0.4160]], dtype=torch.float64) = Variable(data=tensor([[0.4160]], dtype=torch.float64), name=Variable__680).data
E            +    where Variable(data=tensor([[0.4160]], dtype=torch.float64), name=Variable__680) = <theseus.embodied.collision.eff_obj_contact.EffectorObjectContactPlanar object at 0x7f813704edd0>.eff_radius

theseus/embodied/collision/tests/test_eff_obj_contact.py:180: AssertionError


```

Reply:
Ah, right, it's because of the `view(-1, 1)` we added. In this case, let's stick to compare that they are close in value as you had initially.

Reply:
@luisenp Then can you take a look at PR. I've already committed the changes